### DISCUSSION TITLE
Recommendations on a RS485-Wifi server/gateway

### DISCUSSION THREAD

**Discussion by MaxVonEvil** [tagyoureit/nodejs-poolController#833] [RS485]
Hi all,
**TLDR question; Is there a readymade RS485-to-Wifi gateway/server (other than a pi-0) which is compatible with a Nixie RS485 network port? Is Nixie able to talk to anything non-SoCat over the network?**

Background: I used a direct RS485 cable connection for Nixie/NJSPC at our old place, yet the layout of the new house does not lend itself well to cable, so I'm looking at wireless options.

As I have a good wifi-mesh covering the backyard, I would like to get a RS485-Wifi gateway known to work with Nixie's network port option. I understand one option is putting a Pi Zero by the pump, running SoCat, yet I've had outdoor Pi's die on me before, due to the seasonal hot weather here in southern Nevada.

For wireless RS485 in general, I can see there's been some discussion of the [Elfin EW-11](https://www.amazon.com/Smallest-Elfin-EW11-Elfin-EW11-0-Wireless-Networking/dp/B0833WD8H8?th=1), however I'd like to avoid having RS485 hardware in the controller end, as end state goal is having Nixie running in a docker container on a Synlogic NAS, instead of on a Pi.

I was looking at equipment such as [this one](https://www.amazon.com/Rail-Mount-Converter-MultiTransparent-Transmission-Configuration/dp/B0B54TGG1R/ref=sr_1_3?crid=200PD26SH30A&keywords=socat+rs485+wifi&qid=1690672869&sprefix=socat+rs485+wifi%2Caps%2C154&sr=8-3), or any of these [others](https://www.amazon.com/s?k=socat+rs485+wifi&crid=200PD26SH30A&sprefix=socat+rs485+wifi%2Caps%2C154&ref=nb_sb_noss) on Amazon, yet not sure if they are compatible with Nixie?

What caught my eye is that the Nixie port setup, where the description says 'Socat, ETC'
![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/a97feba0-cea0-4896-bda9-3d01d6b07644)

So I guess my question is, what's the ETC for? What else can Nixie talk to over the network. To illustrate what I'm hoping to do is this. Question is, what can I put in front of the pump?

![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/79d24e69-ddcb-480d-bf58-f4c83b764ce7)

PS: I suppose a slightly alternate option is one of these com-port servers where you install a virtual com port driver on the controller side, however if there's a way to avoid having to install anything driver-wise on the host-os of the NAS it would make me sleep better :)

Thoughts? Any hardware recommendations? Thank you!

**Comment by rstrouse** [comment]
Lets start with I don't use any of the mentioned adapters.  From the brief read over of most of them, they are the same concept.  Including the EW11.  The device itself is a TCP server running on an assigned port that relays the RS485 data over the socket.   This is the way all of them work and there is no adapter in the pi for its use.

**Reply by CodeSlinger69** [reply]
The EW11 was great for me when I was using njsPC as a front-end to my EasyTouch. I had njsPC running in a VM on my PC in the house, and the EW11 sending RS485 signals over Wifi. Sounds like what you want to do too.

If you start getting into a full Nixie build, however, you're gonna need a RPi at the pool equipment pad. There are a lot of wires that need to be connected to the RPi, not just the RS485 bus. I am in Phoenix and have the same heat problem as you - I'm currently exploring enclosures and fans to keep the thing cool. My write up is [here ](https://github.com/tagyoureit/nodejs-poolController/discussions/801)if you are interested in this path.

**Reply by MaxVonEvil** [reply]
Thanks for the quick replies gents,

@rstrouse, it looks like I might be doing some experimenting on some of those units (hooray for Amazon's 30 day return policy, lol) and see what works. Btw, I might have messed up on the terminology as I thought Nixie was the new name for NJSPC.

@CodeSlinger69 Thanks for the suggestions, I'll certainly be following your fan/cooling adventures as my current ESP setup may need some help if the summers keep getting hotter. I'll share my thoughts in your discussion.

- -

Just for some some background, my pool adventures started at a rental house where a) I could only do minimally invasive changes which needed to be fully reversible+removable when we eventually bought our own place. As the rental was equipped with a noisy singlespeed 1kW pump, I bought my own Intelliflow VS and plumbed it in with pipe unions, so I could put the old clunker back when we eventually moved post-Covid.

I've been on Home Assistant for years, building my own custom dashboards, ESP32 devices and all the smarts of the house is in NodeRed, including pump, lights and valve actuator controls for the pool. While the capabilities of NJSPC indeed are impressive, I never had clorinators or any proprietary pool control panel - I guess I didn't know what I was missing. Long story short, I've only been using NJSPC to control my VS pump for a long time and the ESP based controller I built on the side just ran the relays and a DS18B20 stainless temperature probe mounted inside the pump basket.

A quick note on the last item, the pentair pumps feature those threaded drain plugs on the side of the pump basket, I bought a spare plug [here](https://www.amazon.com/gp/product/B004VTH78E/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1), drilled a hole through it and pressure mounted the sensor in it, no sealant required.

**Comment by MaxVonEvil** [comment]
Well, so that didn't work ... :)

For my curiosity and your reading entertainment, I ordered [one of these](https://www.amazon.com/dp/B0B54TGG1R?psc=1&ref=ppx_yo2ov_dt_b_product_details) kwang-bong brand RS485-wifi servers on Amazon, as I wanted per the thread above to explore alternatives to the Pi-Zero/SoCat route.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/e01015c3-8090-4d15-b14e-7332c3e5fa71)

What made me pull the trigger on the purchase was it has a transparent-mode, where supposedly all traffic is relayed to a IP port on either UDP or TCP, dealer's choice. Seamed like a good match with NJSPC

While the [setup documentation](https://www.waveshare.com/w/upload/6/61/RS485-TO-WIFI-ETH-User-Manual-EN.pdf) reads like the lovechild between a Mandarin toaster-oven manual and a migraine attack, I managed to get the device onto my network and configurable via it's web interface. Hooked it up to a powersupply and the pump's data port, configured the UART:

![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/ffbeaeec-cdc0-48d2-8282-30c01d89db33)

Configured the network side of the gadget, seems straight forward:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/c3e58bbf-fa99-45ae-bb00-430ec9a4b62f)

On the NJSPC side, I configured a RS485 network port matching the settings on the gadget:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/23d8496b-5e15-49e4-9df2-30c5b248f8cb)

I got as far as it actually connected and NJSPC says the port is open as shown above, yet no data comes through. I know from experience using cable previously, that once there's comms are established, the pump display blanks out with a DISPLAY INACTIVE message, locking out local control. That does unfortunately not happen.

I've tried some of the other modes the serial server offers, yet none of them yields any different result:

![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/64dc8611-ec2f-4056-bcf3-3727304c49ac)

![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/60d7b9e6-d558-4e31-b72a-e316676df4c7)

So, hands down I think I've given this server the time of day, to reasonably determine if it might work with NJSPC. While I think I'm at the end of the rope with this thing and about to return it next week, I would appreciate a note if you think there's something obvious I might have missed. 

Alternatively if there's other industrial hardware out there known to be Socat compatible, i.e. NJSPC can speak to it - I would not mind giving it a try.

**Reply by rstrouse** [reply]
On the njsPC side I assume you added the pump to the configuration and set up a circuit.  These type of devices are great if you can decipher the settings and get them set up.   Unfortunately the documentation if it exists can be deceptive.

**Reply by CodeSlinger69** [reply]
I still don't understand what you are trying to do here. If all you want to do is run njsPC on a box inside your house, as a front end to the OCP, then get an EW11. Its proven and works.

**Reply by MaxVonEvil** [reply]
@rstrouse, correct - that's exactly what I did, yet so far no love with this cursed device. I got as far  today as the pump display went into DISPLAY INACTIVE state, so it must have received *something*, however nothing is coming back the other way. I'll need to doublecheck wiring etc.  

@CodeSlinger69, thank you - yet correct me if I am wrong, the EW11 needs a transceiver device in both ends, right? I am attempting to avoid that setup, as my end-state goal is to run NJSPC inside docker on my NAS, which runs a few other critical systems. See [graphic](https://user-images.githubusercontent.com/19863636/257036228-79d24e69-ddcb-480d-bf58-f4c83b764ce7.png) in post above. As I can't attach a RS485 transceiver to my NAS, I'm leaning towards a network server such as the one I gave a shot above. If you're telling me that the EW11 is able to do that, run a socat server in one end, I'm all ears.

**Reply by CodeSlinger69** [reply]
> @CodeSlinger69, thank you - yet correct me if I am wrong, the EW11 needs a transceiver device in both ends, right? I am attempting to avoid that setup, as my end-state goal is to run NJSPC inside docker on my NAS, which runs a few other critical systems. See [graphic](https://user-images.githubusercontent.com/19863636/257036228-79d24e69-ddcb-480d-bf58-f4c83b764ce7.png) in post above. As I can't attach a RS485 transceiver to my NAS, I'm leaning towards a network server such as the one I gave a shot above. If you're telling me that the EW11 is able to do that, run a socat server in one end, I'm all ears.

The EW11 is a bridge between your RS485 bus and your WiFi network. Its exactly the thing you describe in the title of this thread. One end of the EW11 has 4 wires that you connect to the RS485 bus on your OCP. The other end of the EW11 has an antenna - you configure the EW11 to connect to your WiFi network.

From there, the OCP's RS485 bus is addressable over your local network, and njsPC natively supports this kind of connection.

**Reply by MaxVonEvil** [reply]
Okay, great - initially when I had a look at it, it only seemed as it was a hardware-hardware bridge, meaning it needed something in both ends, yet if it's not the case I'll give it a go. Thanks for clarifying!

**Reply by rstrouse** [reply]
You do not need two devices for the EW11.

**Comment by MaxVonEvil** [comment]
Yes, I found a cheaper kit with only one EW11A device on [Amazon](https://www.amazon.com/dp/B08G8LFFGN?psc=1&ref=ppx_yo2ov_dt_b_product_details). It was super quick to set up and worked immediately. This accomplishes what I set out to do; getting wireless comms up and getting NJSPC off the Pi and onto the NAS.

The only snafu I ran into, was getting the paths right in the docker-compose.yaml file, for running on top of my Synology NAS. If others are interested in going the same route, [look here](https://www.wundertech.net/how-to-use-docker-on-a-synology-nas/) for the nitty-gritty, but here's what my file looks like.

```
services:
  poolcontroller:
    container_name: poolcontroller
    restart: always
    user: node:dialout
    ports:
      - "4200:4200"
    volumes:
    - /volume1/docker/njspc/poolcontroller/data:/app/data
    - type: bind
      source: /volume1/docker/njspc/poolcontroller/config.json
      target: /app/config.json
    image: msmi/nodejs-poolcontroller
  poolcontroller-dashpanel:
    restart: always
    container_name: poolcontroller-dashpanel
    ports:
      - "5150:5150"
    volumes:
    - type: bind
      source: /volume1/docker/njspc/poolcontroller-dashpanel/config.json
      target: /app/config.json
    image: msmi/nodejs-poolcontroller-dashpanel
    depends_on:
      - poolcontroller
```

Before uploading the compose file, I SAMBA'ed into the Pi, grabbed the /data folder and copied it into the docker file structure using the Synology Filestation: 

![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/69cb81a3-670e-427e-bdd2-7ab090061f7d)

And here is what the final setup looks like;
![image](https://github.com/tagyoureit/nodejs-poolController/assets/19863636/0e79713e-e85e-4def-bd72-4fae1535c1aa)

now I just need to stick it all in a nice outdoor box ;)

**Reply by MyTek** [reply]
Hey pretty funny I just found this and I think it's basically what I'm looking for (I'm also in southern Nevada haha).  I'm looking to hook up a pool heater and maybe a switch for the power to my pool motor.  Right now I have a busted heater that the previous tenants bypassed and a single speed pool pump with a simple power on/off switch.  

I'm looking at what kind of pool heater I can use and was thinking of a pentair mastertemp 125.  I'm curious on what to get for the pool pump control though.  How do you manage to connect more than one device to the ew11?  Would I need two ew11's?  Any suggestions on a simple on/off switch for the pump to connect it to a setup like this?

**Reply by MaxVonEvil** [reply]
Small world, as I happened to have a MasterTemp 125 at our old place here in Henderson. Should be said I never found any data interface on that heater, so I just used the fireman's switch (see fig 24 in the manual) to turn it on/off with a remote controlled relay. 

How you go about that, depends on how you want to control said relay. NJSPC can do it with it's relay manager. Some people have pool controller gear in their backyards which they want to have an interface to. In my instance I started out with a 100% manual system.

I may be a borderline heretic in this place 😆 as I only use NJSPC as an interface to my pump, the rest I got under control with HomeAssistant. In the case of the pool heater and my valve actuators I have an ESP32-S2 Feather from AdaFruit flashed with Tasmota, hooked up to a 8x relay board. 

Should be said that I added a DS18B20 temp sensor to the pump basket, so I have a continuous reading.  I tied that in with the heater, which I set on its panel to its max. temp of 104, using the fireman's switch to turn it on/off automatically using a temperature control loop in NodeRed. That way I can set the spa and/or pool to any specific temperature I want.

I know it all might somewhat...involved, yet I started building in the other end of the boat, however I'm happy with 
things, especially now got the EW11 
talking to the pump

To circle back to your question, unless you buy a heater which comes with a data port, you really have no use for a EW11, just a relay.

**Reply by MyTek** [reply]
Hey thanks for the info!  I figured since the mastertemp had an RD485 port it would interface with something out there.  I saw a couple repos out there and im just starting my journey here.  I do devops and web development by trade so I think I can pick it up.  I guess I just need the interface.  

My question really is once the ew11 is hooked up, how does the interface software find them?  Like you just give the config the ew11 ip address and your good?  I have a server so I can spin something up.

Also, I haven't bought it yet so if you had a recommendation for a heater that interfaces let me know.  It's a small pool though.  Maybe 5000 gallons.

**Reply by MaxVonEvil** [reply]
We'll, the NJSPC software is relatively easy to set up, I prefer running it on Docker myself. Once you have the web interface (poolcontroller-dashpanel) running you configure a network port and exactly as you say, you provide the IP and port (default tcp/8899) of the EW11. You also need to know the serial config of the heater (may be 9600 N 8 1, guessing here) and set that up in the EW11.

As for heaters, never really bought one as the new house came with an even older model, which looks like it was made in Russia and eats plutonium fuel rods for breakfast, lol. But hey it's got a switch, so I can control it business as usual.

**Reply by MyTek** [reply]
Well thanks for all the info!  I'll reach back out if I change this up a bit and get something working.

**Reply by MyTek** [reply]
Oh one last thing.  Do you have a link to the ew11 you bought?  I see a bunch of ones on ebay and aliexpress but the prices are all over the place so I don't want to get screwed!

**Reply by MaxVonEvil** [reply]
Yep, there is a link above. Look for linked word 'Amazon', it's listed at around 20 bucks

**Reply by MyTek** [reply]
Well how about that.  Thanks!

**Reply by dsarvadi** [reply]
This is a very helpful discussion of some details on using WIFI to connect to RS485/Pentair Easytouch.  But I can't find the EW11 on amazon anymore, so I bought a Waveshare RS232/485 to WiFi and Ethernet Converter Serial Server, Modbus Gateway, MQTT Gateway,Serial Port to HTTPD Client, that Allows Serial Port to WiFi, Serial Port to Ethernet, Ethernet to WiFi.  I'll try to use the yaml script above to integrate, but was looking for any insight from anyone who has tried this with a different rs485 to wifi adapter.  Thanks for the help in advance.

**Reply by tagyoureit** [reply]
Looks like the EW11 is EOL and the EW11a is the current version.  [Newegg link](https://www.newegg.com/p/2S7-01Z1-00EG9?srsltid=AfmBOooDCN6ymWTFY1bcV_8TFXQ8A1h4P7zmyftTR-_lXBZSr6vtnAeh)

**Reply by dsarvadi** [reply]
The waveshare is directly connected to the Screenlogic remote in my office.  I'm getting an IP address and the lights on the device seem to be operating correctly.  Not sure of the connections are correct but they match the connections on the Screenlogic connection between the interior device and the antenna.  See attached image.
![20260309_104428](https://github.com/user-attachments/assets/39a1ac33-ba8c-4688-a7ba-fb58b6f0bb63)
I don't know enough of java programming to edit the config.json to correctly identify the Waveshare device and connect to the RS485.  I see the RS485 port defined as "/dev/ttyUSB0" and the nethost as raspberrypi.   I am connecting the system to the IP address of the waveshare through an Ubuntu machine.  I assume these two lines in the file need to be edited and I have searched the web for simple instructions on this but so far no luck. 
 
In addition, my screenlogic device is identified in the file correctly with the Pentair ID and the correct password to access the data stream.  I installed nodejs-poolController following the instructions in tagyoureit's github page.  When I run npm install, I get the foillowing:
"4 high severity vulnerabilities

Some issues need review, and may require choosing
a different dependency."

npm audit produces the following: 

"# npm audit report

ip  *
Severity: high
ip SSRF improper categorization in isPublic - https://github.com/advisories/GHSA-2p57-rm9w-gvfp
No fix available
node_modules/ip

minimatch  <=3.1.3
Severity: high
minimatch has a ReDoS via repeated wildcards with non-matching literal in pattern - https://github.com/advisories/GHSA-3ppc-4f35-3m26
minimatch has ReDoS: matchOne() combinatorial backtracking via multiple non-adjacent GLOBSTAR segments - https://github.com/advisories/GHSA-7r86-cg39-jmmj
minimatch ReDoS: nested *() extglobs generate catastrophically backtracking regular expressions - https://github.com/advisories/GHSA-23c5-xmqv-rm74
No fix available
node_modules/grunt/node_modules/minimatch
  grunt  >=0.4.0-a
  Depends on vulnerable versions of minimatch
  node_modules/grunt
    grunt-banner  *
    Depends on vulnerable versions of grunt
    node_modules/grunt-banner

4 high severity vulnerabilities
"
No fix available for minimatch vulnerability!  And I can't find one for the grunt error either.
 When I start the program, npm start, I get this error "node_modules/socket.io/dist/index.d.ts:1:8 - error TS1192: Module '"http"' has no default export.

1 import http from "http";
         ~~~~


Found 1 error in node_modules/socket.io/dist/index.d.ts:1"

Installing nodejs-poolController-dashpanel produces errors and npm audit produces this report:
"# npm audit report

ip  *
Severity: high
ip SSRF improper categorization in isPublic - https://github.com/advisories/GHSA-2p57-rm9w-gvfp
No fix available
node_modules/ip
  node-ssdp  >=1.0.1
  Depends on vulnerable versions of ip
  node_modules/node-ssdp
    nodejs-poolcontroller-dashpanel  *
    Depends on vulnerable versions of node-ssdp
    node_modules/nodejs-poolcontroller-dashpanel

3 high severity vulnerabilities

Some issues need review, and may require choosing
a different dependency."

Another vulnerability that has no fix.  I'm at a dead end.  I'm still a NOOB so have no idea where to go next.   Any help appreciated.

**Reply by tagyoureit** [reply]
You don't want to have the screenlogic connected if you have direct RS485 connectivity.  The Screenlogic is very limited compared to the direct bus.

You want to have the comms port setup as NETWORK and nethost as RASPBERRYPI.

<img width="567" height="457" alt="image" src="https://github.com/user-attachments/assets/49c4b72c-f50c-4dae-9853-5b6cac2071d6" />

You don't need to edit the config.json manually.  Everything can be done through the UI.

And never, never get the gremlins wet.  I mean... never, never run any `npm audit` commands.  It will lead you to a bad place.

**Comment by dsarvadi** [comment]
Thanks for the response.  I'm not running on RPi, but an ubuntu laptop. 
The RS485 is connected to the screenlogic interior wifi connector and 
has a local IP address.  I addition the screenlogic wifi has it's own IP 
address.  I think this section needs to have a reference to a dev/tty ip 
address instead of dev/tty/USB.Sorry for the unusual setup, but I"m 
hoping to have the tablet for the household to use the dashboard in 
homeassistant.

"comms": {
       "type": "NETWORK",
       "portId": 0,
       "enabled": true,
       "rs485Port": "/dev/ttyUSB0",

I tried looking up how to program this, but was unsuccessful. Any 
additional help appreciated.

703-753-9978
703-906-1177 (C)
***@***.***

**Reply by tagyoureit** [reply]
You don’t use a /dev/tty* path for a network RS485 bridge. Those paths are only for a local USB/serial adapter on the machine running nodejs-poolController.

For a Waveshare (or similar) serial server:

1. In the pool controller UI, set the RS485 port to the network / Socat-style connection (not ScreenLogic, not local serial).
1. Set Host to the Waveshare’s IP address (or a hostname that resolves to it). The default raspberrypi is just an example; on an Ubuntu laptop you’d use something like 192.168.1.xxx.
1. Set Port to the TCP port the Waveshare is using in transparent TCP server mode (whatever its web UI shows—must match exactly).
1. On the Waveshare, baud/data bits/parity/stop bits must match the Pentair side (often 9600 8N1 for classic gear—confirm against your equipment/Waveshare docs).

So: the IP goes in Host, not in a dev/tty string. If rs485Port still shows /dev/ttyUSB0 in config.json while you’re on network mode, that’s often a leftover label; the active connection is Host:Port once network mode is enabled. Prefer changing it in the UI so the config stays consistent.

Important caveat about your wiring: nodejs-poolController’s full RS485 support assumes a transparent bridge to the real Pentair bus (same idea as an EW11 on the OCP bus). The link between the ScreenLogic indoor unit and the wireless antenna is not necessarily the same as “the whole panel bus,” and mixing ScreenLogic credentials in config with a network serial port pointed at a device wired only to ScreenLogic can be confusing. If the goal is “full” panel control, the usual approach is one serial server on the equipment RS485 bus, or use the ScreenLogic port type alone (no Waveshare in the middle) and accept the more limited ScreenLogic data model—as @tagyoureit noted.

Separate issues you mentioned: npm audit noise is common and not a blocker for many installs. The socket.io / http default export TypeScript error usually means the Node/TypeScript toolchain doesn’t match what the project expects—worth opening a new issue with Node version, OS, and exact npm start output so someone can point you at the supported versions.

**Comment by dsarvadi** [comment]
Thanks. Let me work through this. Sent from my Galaxy
-------- Original message --------From: tagyoureit ***@***.***> Date: 4/1/26  9:33 PM  (GMT-05:00) To: tagyoureit/nodejs-poolController ***@***.***> Cc: dsarvadi ***@***.***>, Comment ***@***.***> Subject: Re: [tagyoureit/nodejs-poolController] Recommendations on a RS485-Wifi server/gateway (Discussion #833) 
You don’t use a /dev/tty* path for a network RS485 bridge. Those paths are only for a local USB/serial adapter on the machine running nodejs-poolController.
For a Waveshare (or similar) serial server:

In the pool controller UI, set the RS485 port to the network / Socat-style connection (not ScreenLogic, not local serial).
Set Host to the Waveshare’s IP address (or a hostname that resolves to it). The default raspberrypi is just an example; on an Ubuntu laptop you’d use something like 192.168.1.xxx.
Set Port to the TCP port the Waveshare is using in transparent TCP server mode (whatever its web UI shows—must match exactly).
On the Waveshare, baud/data bits/parity/stop bits must match the Pentair side (often 9600 8N1 for classic gear—confirm against your equipment/Waveshare docs).

So: the IP goes in Host, not in a dev/tty string. If rs485Port still shows /dev/ttyUSB0 in config.json while you’re on network mode, that’s often a leftover label; the active connection is Host:Port once network mode is enabled. Prefer changing it in the UI so the config stays consistent.
Important caveat about your wiring: nodejs-poolController’s full RS485 support assumes a transparent bridge to the real Pentair bus (same idea as an EW11 on the OCP bus). The link between the ScreenLogic indoor unit and the wireless antenna is not necessarily the same as “the whole panel bus,” and mixing ScreenLogic credentials in config with a network serial port pointed at a device wired only to ScreenLogic can be confusing. If the goal is “full” panel control, the usual approach is one serial server on the equipment RS485 bus, or use the ScreenLogic port type alone (no Waveshare in the middle) and accept the more limited ScreenLogic data model—as @tagyoureit noted.
Separate issues you mentioned: npm audit noise is common and not a blocker for many installs. The socket.io / http default export TypeScript error usually means the Node/TypeScript toolchain doesn’t match what the project expects—worth opening a new issue with Node version, OS, and exact npm start output so someone can point you at the supported versions.

**Comment by dsarvadi** [comment]
Thanks so much for you help.  I've got things running, but haven't quite 
connected to the RS485.  Dashpanel is running but no data of course, but 
I haven't been able to figure out how to access the UI for NJSPC.  I 
have the localhost IP, but haven't been able to figure out which port to 
access.  Silly noob issue.

Here's the error message I get: "error: Net connect (socat) connection 0 
error: Error: connect ECONNREFUSED 192.168.XXX.XXX:8899.  Retry in 10 
seconds
[4/4/2026, 1:26:08 PM] info: Net connect (socat) 0 closed due to error: 
192.168.XXX.XXX:8899,  I suspect it has something to do with the socat 
error you mentioned below. but haven't had time to go research it. 
Thoughts? Fixes?

Anyway, many thanks for the help.  It's been a struggle but I feel like 
I'm getting there.

David Sarvadi
28505 Azzili Way
Bonita Springs FL 34135
239-314-0305
703-753-9978
703-906-1177 (C)
***@***.***

**Reply by tagyoureit** [reply]
DashPanel is the UI for NJSPC.  Follow these arrows to get to the network settings to connect to the waveshare.

<img width="773" height="494" alt="2026-04-07_17-41-16" src="https://github.com/user-attachments/assets/4119fda2-8732-4365-9be1-cd8cc62c3ff9" />

<img width="633" height="535" alt="2026-04-07_17-41-50" src="https://github.com/user-attachments/assets/1f39509c-a2c7-4752-b1de-5f4a16d87c63" />

**Comment by dsarvadi** [comment]
Got it.  Thanks.

David Sarvadi
28505 Azzili Way
Bonita Springs FL 34135
239-314-0305
703-753-9978
703-906-1177 (C)
***@***.***
